### DISCUSSION TITLE
Problem tunneling between two servers

### DISCUSSION THREAD

**Discussion by Takashi-YOL** [xjasonlyu/tun2socks#168] [Q&A]
I am a beginner, I was trying to tunnel between my computer to `VPS1` (which has restricted internet due to Iran government policy) and VPS2 (From a Europan provider). Just rented an Ubuntu 20 VPS server from the Iranian provider (`VPS1`). I tried installing and configuring `shadowsocks `and `v2ray `and use `tun2socks `using these commands:

`nano /etc/resolv.conf`

`nameserver 8.8.8.8`

`nameserver 8.8.4.4`

`apt update`

`apt install shadowsocks-libev tmux`

`apt install vsftpd`

`apt install unzip`

`wget https://github.com/xjasonlyu/tun2socks/releases/download/v2.4.1/tun2socks-linux-amd64.zip`

unzip tun*
I used this config for tun2socks (https://github.com/xjasonlyu/tun2socks/wiki/Examples):

`ip tuntap add mode tun dev tun7`

`ip addr add 198.18.0.1/15 dev tun7`

`ip link set dev tun7 up`
and then :

`wget https://github.com/shadowsocks/v2ray-plugin/releases/download/v1.3.2/v2ray-plugin-linux-amd64-v1.3.2.tar.gz`

`tar xf v2ray*`
Then I used this command:

`ufw allow 80`
I created this script as PEPE.json:

```
{
   "server":"0.0.0.0",
   "mode":"tcp_and_udp",
   "server_port":80,
   "password":"passsssssssss",
   "timeout":60,
   "method":"chacha20-ietf-poly1305",
   "nameserver":"8.8.8.8",
   "plugin":"v2ray-plugin_linux_amd64",
   "plugin_opts":"server"
}
```
I ran the script :

ss-server -c PEPE.json -i tun7
and finally:

```
./tun2socks-linux-amd64 -device tun7 -proxy http://X.X.X.X:22
I put my VPS1 server IP and port as X.X.X.X
```

In the end, I could receive no connection at all. What did I do wrong? Do I have to tunnel between `VPS1 `to VPS2 ( The Europian VPS) using a specific command? I did replace `X.X.X.X` with the `VPS2 `but still, nothing happened.

I would appreciate any help to solve my issue.

There is a video here showing the path I followed:

https://t.me/CyberNigga2/996

**Comment by xjasonlyu** [comment]
Sorry that I didn’t understand this commands were executed on which VPS. 

From what I can tell, tun2socks should only be used  in your local computer, and your VPSs should be connected/tunneled by V2RAY/SS proxy. Then you expose a SS port to accept connection from your local PC (tun2socks).

**Reply by Takashi-YOL** [reply]
Hi
Thank you for your reply
All of these were executed on the VPS1 (Iranian VPS server which has restricted access)

**Reply by Takashi-YOL** [reply]
To bypass censorship, I have to get connected to an offshore VPS. My local computer and Iranian VPS (VPS1) have limited access to the internet. VPS 2 is located in Europe, for example. What should I do to connect my local computer to the VPS2 ? 
Is it possible for you to guide me establish a connection?
